### PR TITLE
[M] Initialize API routes (#37)

### PR DESCRIPTION
**Pull request by burhan-syed**
Setup /products route as example

### CODE REVIEW

**Review by justcord10 — CHANGES_REQUESTED** [review]
Changes to make: add a note for the others in the server/router to tell them explicitly where to put their paths. also might as well uncomment where you require the other controllers in the server/controllers/index.js file to save a pull request later down the line. 
Questions i have, should the require('dotenv') statement live in the models/index.js file or should we put it in the server/index.js file instead so that we add in PATH into the .env file?
Also do you want to walk me through your models/index.js files so that i can better understand the code?
other than that README changes look good, good updating the gitignore file to ignore .env files, example.env looks fine maybe consider throwing some dummy strings("FILL_ME_IN") in there for when people copy the files, good checks to ensure people set up their .env files properly, good router setup and middleware usage in the server/index.js file

**Comment by burhan-syed** [comment]
> Changes to make: add a note for the others in the server/router to tell them explicitly where to put their paths. also might as well uncomment where you require the other controllers in the server/controllers/index.js file to save a pull request later down the line.  

Changes made

>  should the require('dotenv') statement live in the models/index.js file or should we put it in the server/index.js file instead so that we add in PATH into the .env file?  

Environment variables are only being used in the model to call the external products API. The dotenv is loaded once there and won't be used anywhere else so I don't think it needs to be in the server/index.js file. Not sure what is meant by adding PATH into the .env file. 


> walk me through your models/index.js files so that i can better understand the code  

For future reference, this file is assuring the required environment variables are defined then exporting a `callAPI` function. The `callAPI` function accepts a request object and returns the promise from an axios get request with the required Authorization key. The request object should be passed in from the controller code so the get request can be made to the corresponding API endpoint.    

As an example: 
If our server is running on localhost:3000 and the external products api is on https://example.com. 
A GET request to localhost:3000/products/productNum1/styles
will be proxied by the server to https://example.com/products/productNum1/styles

File: server/index.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 const express = require('express');
 const path = require('path');
+const router = require('./router');
 
 const app = express();
+app.use(express.json());
+app.use('/', router);
 app.use(express.static(path.join(__dirname, '../client/dist')));
 
 const PORT = process.env.PORT || 3000;
```
Comment (justcord10) [inline server/index.js:10]: right here we reference process.env.PORT and it is the highest level in our server code so the .env file should be in here and a PORT variable added to the example.env file.

**Review by justcord10 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -23,6 +23,9 @@ npm install
 
 ## Usage
 
+Look at .env.example and create a .env file with required variables.
+The API Endpoint should not have a trailing slash.
```
Comment (justcord10) [inline README.md:27]: rename to API host for more consistent naming to match example.env changes

File: example.env [hunk 3]
```diff
@@ -1 +1,2 @@
-# TODO
\ No newline at end of file
+GITHUB_TOKEN=YOUR_GITHUB_API_TOKEN
+API_HOST_URI=https://exampe.app.com/api/endpoint
```
Comment (justcord10) [inline example.env:2]: add in a POST variable

**Review by justcord10 — CHANGES_REQUESTED** [review]

**Review by justcord10 — APPROVED** [review]
Okay after that last change looks good, will merge now

**Pull request merged**